Library/Homebrew/brew.sh

```diff
@@ -290,9 +290,22 @@ auto-update() {
       fi
     fi
 
-    # Skip auto-update if the Homebrew/brew repository has been checked in the
-    # last $HOMEBREW_AUTO_UPDATE_SECS.
     repo_fetch_head="${HOMEBREW_REPOSITORY}/.git/FETCH_HEAD"
+    if [[ -n "${HOMEBREW_AUTO_UPDATE_CORE_CLONES}" ]]
+    then
+      # We might have done an auto-update recently, but not a core clone auto-update.
+      # So we use the core clone FETCH_HEAD instead.
+      if [[ -d "${HOMEBREW_CORE_REPOSITORY}/.git" ]]
+      then
+        repo_fetch_head="${HOMEBREW_CORE_REPOSITORY}/.git/FETCH_HEAD"
+      elif [[ -d "${HOMEBREW_CASK_REPOSITORY}/.git" ]]
+      then
+        repo_fetch_head="${HOMEBREW_CASK_REPOSITORY}/.git/FETCH_HEAD"
+      fi
```

Comment:
Totally fine as this as-is but I'm wondering if it always makes sense to pick them based on existence rather than e.g. `bump` needs core and cask, `bump-formula` just needs core, `bump-cask` just needs cask? Don't want to overcomplicate this, just a thought.

Reply:
Possibly, though auto-update will currently update both so it sounds like you maybe are asking for separate `HOMEBREW_AUTO_UPDATE_CORE_TAP` and `HOMEBREW_AUTO_UPDATE_CASK_TAP`?

Reply:
Yeh maybe although that may be overkill. Feel free to tell me it is, I don't care much!

Reply:
It does make it a little more complicated but I think it is worth it as it could save minutes. I've made this change.